pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -666,6 +666,9 @@ func validateValidatingAdmissionPolicySpec(meta metav1.ObjectMeta, spec *admissi
 			allErrors = append(allErrors, field.Required(fldPath.Child("matchConstraints", "resourceRules"), ""))
 		}
 	}
+	for i, matchCondition := range spec.MatchConditions {
+		allErrors = append(allErrors, validateMatchCondition(&matchCondition, fldPath.Child("matchConditions").Index(i))...)
```

Comment:
We do expect parameters for validation. I'll need to add some plumbing.

Reply:
that'll make the API doc on the MatchCondition expression field interesting to write

Reply:
Yeah, maybe for now only document it in `matchConditions` since that is bound directly to VAP types.

Reply:
Is there a reason you used the singular `validateMatchCondition` here rather than `validateMatchConditions`? This is missing the name uniqueness & max-64 conditions checks.

Also, should we reuse the `opts.IgnoreMatchConditions` setting for this?

Reply:
> Also, should we reuse the opts.IgnoreMatchConditions setting for this?

If we do that, make sure we re-validate if they change `paramKind` to/from nil (affects whether the conditions can access a `params` var)

Reply:
The boolean just controls whether to validate the matchConditions field or not.

For webhooks, that is populated based on whether or not they have changed in an update. You don't have to wire it to a feature gate at all

Reply:
Oh, I think I misunderstood the match conditions bit, you want to save on the compute if nothing has changed... gotcha. Will add.

Reply:
Added

Reply:
> you want to save on the compute if nothing has changed

It also makes it much easier to change validation constraints in the future (e.g. if we relaxed the max number of matchConditions).